sdk/core/core-amqp/src/retry.ts

```diff
@@ -217,16 +209,16 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
         err
       );
       let targetDelayInMs = config.delayInMs;
-      if (config.retryPolicy === RetryPolicy.ExponentialRetryPolicy) {
+      if (config.mode === RetryMode.Exponential) {
         let incrementDelta = Math.pow(2, i) - 1;
         const boundedRandDelta =
           config.delayInMs * 0.8 +
           Math.floor(Math.random() * (config.delayInMs * 1.2 - config.delayInMs * 0.8));
         incrementDelta *= boundedRandDelta;
 
         targetDelayInMs = Math.min(
-          config.minExponentialRetryDelayInMs + incrementDelta,
-          config.maxExponentialRetryDelayInMs
+          defaultMinDelayForExponentialRetryInMs + incrementDelta,
```

Comment:
`defaultMinDelayForExponentialRetryInMs` is to be removed as well

Reply:
Per @bterlson's comments, the retryDelayInMs is to be used as minimum correct? This has value of 30 seconds, whereas for exponential retry we had minumum as 3 seconds.
Since this issue/PR is only for renaming, refactoring, I assume we didn't want to make changes to the logic as such.

Reply:
The `retryDelayInMs` is indeed getting used as the minimum. If you look at the math, the variable `incrementDelta` has the exponentially increasing value we want for the delay.

As per https://github.com/Azure/azure-sdk/pull/534, we no longer need to have a separate min value to which something needs to be added. Therefore, no need for either the configurable minimum (`config.minExponentialRetryDelayInMs`) or the default minimum (`defaultMinDelayForExponentialRetryInMs`)